mobile/src/components/course/components/edit-course-category/edit-course-category.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { FC } from 'react';
+
+import { AppScreenName, DataStatus } from '~/common/enums/enums';
+import {
+  CategoryGetAllItemResponseDto,
+  CourseGetResponseDto,
+} from '~/common/types/types';
+import {
+  BackButton,
+  Button,
+  Dropdown,
+  Spinner,
+  View,
+} from '~/components/common/common';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppNavigate,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useFocusEffect,
+} from '~/hooks/hooks';
+import { coursesActions } from '~/store/actions';
+import { courseUpdateCategory as courseUpdateCategoryValidationSchema } from '~/validation-schemas/validation-schemas';
+
+import { DropDownPayload } from './common/common';
+import { getDefaultValueforDropDown } from './helper/helpers';
+import { styles } from './styles';
+
+const EditCourseCategory: FC = () => {
+  const navigation = useAppNavigate();
+  const dispatch = useAppDispatch();
+
+  const { course, dataStatus, categories } = useAppSelector(
+    (state) => state.courses,
+  );
+  const currentCategory = course?.courseCategoryId;
```

Comment:
Here you are sure that there will be a course, so you can cast it at the top
```suggestion
  const courseData = course as CourseGetResponseDto
```